.github/workflows/ui-sonarcloud.yml

```diff
@@ -0,0 +1,75 @@
+name: FOLIO UI SonarCloud
+
+on:
+  workflow_call:
+    inputs:
+      sonar-sources:
+        description:
+          A comma-separated list of directories containing code to analyze (no wildcards).
+          See https://docs.sonarsource.com/sonarcloud/advanced-setup/analysis-parameters/#analysis-scope for more.
+        required: true
+        type: string
+      sonar-exclusions:
+        description: A comma-separated list of wildcards to exclude from analysis.
+        required: true
+        type: string
+      bigtest-coverage-report-dir:
+        description: Directory in which Bigtest coverage reports are stored in, with trailing slash. LCOV data should be in "lcov.info".
+        required: true
+        type: string
+      jest-coverage-report-dir:
+        description: Directory in which Jest coverage reports are stored in, with trailing slash. LCOV data should be in "lcov.info".
+        required: true
+        type: string
+    secrets:
+      SONAR_TOKEN:
+        description: Token for performing a SonarCloud analysis
+        required: true # fails the check if the PR is from a fork
+
+jobs:
+  run:
+    name: SonarCloud analysis
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout ${{ github.repository }}
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          show-progress: false # spammy
+
+      - name: Fetch base branches
+        run: git fetch --no-tags ${{ github.server_url }}/${{ github.repository }} +refs/heads/${{ github.event.repository.default_branch }}:refs/remotes/origin/${{ github.event.repository.default_branch }}
+
+      - name: Fetch coverage reports (Bigtest)
+        uses: actions/download-artifact@v4
+        with:
+          name: bigtest-coverage-report
+          path: ${{ inputs.bigtest-coverage-report-dir }}
+        continue-on-error: true # bigtest may be disabled or have failed
+
+      - name: Fetch coverage reports (Jest)
+        uses: actions/download-artifact@v4
+        with:
+          name: jest-coverage-report
+          path: ${{ inputs.jest-coverage-report-dir }}
+        continue-on-error: true # jest may be disabled or have failed
+
+      # It's so long that it's awful to maintain without this
+      - id: format-exclusions
+        name: Format exclusions list
+        run: echo formatted=$(tr -d '[:blank:]' <<< "${{ inputs.sonar-exclusions }}") | tee -a $GITHUB_OUTPUT
+
+      - name: Run SonarCloud scan
+        uses: sonarsource/sonarcloud-github-action@master
```

Comment:
Is a version available? Running against master feels risky.

Reply:
Sure, that's what sonar recommends though. The action has very little/infrequent code changes so idk how much of an issue this would be

Reply:
We can stick to their recommendation. I don't think it's a BP, as a rule, but as long as we know what the BP is and are making a conscious decision to do something else because we got guidance to do something else, that's totally fine. Consider adding a comment like "Yep we're tracking tip-of master 😲 but it's what SC recommends 🤷 ".

Reply:
We could tag it to a version, but I'm not sure it matters much regardless? The action is really just `download sonar-scanner-cli docker image` -> `run sonar-scanner`

Reply:
*cough cough https://github.com/SonarSource/sonarcloud-github-action/pull/67 This is currently preventing any PR from merging in any UI repo ATM 😱

Reply:
I think that's the most reviewers I've ever seen on a PR...

Reply:
Well, all those "reviewers" were just third-party folks uselessly mashing the review button, hence all those checkmarks are gray instead of green. You are right that pinning to a certain action version would not have helped here, but only because this action _also_ follows the bad-"best"-practice of pulling the latest. IOW, we are victim to "a chain is only as strong as its weakest link" here, but I don't think that's an argument against building our own strong links. 

Maybe SonarSource will learn a thing or two from this fustercluck
1. no, relying on master is NOT a best practice 😒 
2. when you publish a new version, actually publish a new version. what SC did here, and what made this problem so very very bad, was replace functioning v5.0.1 with a broken v5.0.1, so 
    1. even people who had pinned to that specific old version still received the new broken code 😱 
    2. there was no way to roll back 😱 😱 😱 
5. there are some very smart and thoughtful folks in the community who can offer [criticism both pointed and constructive](https://github.com/SonarSource/sonarcloud-github-action/pull/67#issuecomment-2060720747) ❤️ 

And if we all implement actual best practices, next time somebody publishes a broken thing we'll avoid it altogether (because _we_ haven't deliberately opted-in) or have an easy way to roll back (because _they_ didn't overwrite an already-published release). ⛓️ 💪